packages/ceramic-doctype-tile/src/tile-doctype-handler.ts

```diff
@@ -71,18 +71,19 @@ export class TileDoctypeHandler implements DoctypeHandler<TileDoctype> {
      * @private
      */
     async _applyGenesis(record: any, cid: CID, context: Context): Promise<DocState> {
-        await this._verifyRecordSignature(record, context)
-        // TODO - verify genesis record
+        await this._verifySignature(record, context)
+
+        const payload = (await context.ipfs.dag.get(record.link)).value
         return {
             doctype: DOCTYPE,
-            content: record.data,
-            metadata: record.header,
+            content: payload.data,
+            metadata: payload.header,
             next: {
                 content: null,
             },
             signature: SignatureStatus.SIGNED,
             anchorStatus: AnchorStatus.NOT_REQUESTED,
-            log: [cid]
+            log: [record.link]
```

Comment:
Both the CID of the jws and the linked block needs to be added to the log I think.

Reply:
👍

Reply:
I'm pretty sure that just **cid** needs to be in the log.

Reply:
that's what I meant. 
* The CID of the JWS
* The CID of the linkedBlock (jws.link)

Reply:
ok